src/px-script/main/datepicker/index.js

```diff
@@ -57,6 +57,10 @@ const _createDatepicker = datepicker => {
     return datepickerObj;
 };
 
+const _destroyDatepickers = () => {
+    _datepickers.forEach(datepicker => datepicker.destroy());
+};
+
 const init = id => {
     if (id) {
```

Comment:
Should previous datepickers be removed regardless when `init` is invoked ? If so move this repeated statement to the top and remove statements below. If not ignore this and add tests for it.

```suggestion
    if (_datepickers.length) { _destroyDatepickers(); }
    if (id) {
```

Reply:
If i move the test to the top of init then old datepickers would be destroyed if you sent a specific ID. Which raises another relevant question; should datepickers be destroyed when you init a single specific ID? Should probably only be done when you init ALL datepickers.

Reply:
On second thought datepickers would not be destroyed on invalid ID due to the test right before. But datepickers should not be destroyed on ID-init anyways.

Reply:
Right now you are destroying all datepickers if invoking `init` with a valid ID. If that should not be the case, you should remove the `_destroyDatepickers();` line from the `if (id)` block, and add tests to make sure they are not destroyed :)

Reply:
Comment to my self: Destroy the datepicker matching the passed ID.